docs/en/swip/SWIP-1.md

```diff
@@ -0,0 +1,160 @@
+5. Virtual databases, caches, and queues conjectured by agents, and also monitored through k8s monitoring, even traffic
+   monitored by service mesh.
+
+All these services have logic connections or are identical from a physical perspective, but currently, they may be just
+similar on name(s), no further metadata connection.
+
+By those, we have a chance to move one step ahead to connect the dots of the whole infrastructure. This means, for the
+first time, we are going to establish the connections among services detected from various layers.
+
+**In the v10, I am proposing a new concept `Service Hierarchy`.** Service Hierarchy defines the relationships of
+existing services in various layers. With more kinds of agent tech involved(such as eBPF) and deployment tools(such as
+operator and agent injector), we could inject relative service/instance metadata and try to build the connections,
+including,
+
+- Agent injector injects the pod ID into the system env, then Java agent could report the relationship through system
+  properties.
+- Rover(eBPF agent) reveals its next iteration forward k8s monitoring rather than profiling. And add the capabilities to
+  establish connections among k8s pods and service mesh srv.
+
+Meanwhile, as usual with the new major version change, I would expect UI side changes as well. UI should have flexible
+capabilities to show hierarchy services from the service view and topology view. Also, we could consider a deeper view
+of the instance part as well.
+
+# Imported Dependencies libs and their licenses.
+
+No new library is planned to be added to the codebase.
+
+# Compatibility
+
+About the **protocol**, there should be no breaking changes, but enhancements only. New query protocols(
+service-hierarchy and instance-hierarchy) are considered to be added, some new fields should be added on things like
+topology query and instance dependencies to list relative services/instances from other layers directly rather than an
+extra query.
+
+About the data structure, due to the new data concept is going to be created, service hierarchy relative data models are
+going to be added. If the user is using Elasticsearch and BanyanDB, this should be compatible, they just need to
+re-run `init-mode` OAP to extend the existing models. But for SQL database users(MySQL, PostgreSQL), this could require
+new tables.
+
+## GraphQL query protocol
+New query protocol `hierarchy.graphqls` is going to be added.
+```graphql
+type HierarchyRelatedService {
+    # The related service ID.
+    id: ID!
+    # The literal name of the #id.
+    name: String!
+    # The related service's Layer name.
+    layer: String
+}
+
+type HierarchyRelatedInstance {
+    # The related instance ID.
+    id: ID!
+    # The literal name of the #id. Instance Name.
+    name: String!
+    # The related instance service's Layer name.
+    layer: String
+}
+
+type ServiceHierarchy {
+    upper: [HierarchyRelatedService!]!
+    lower: [HierarchyRelatedService!]!
+}
```

Comment:
Let's try a complete hierarchy for service. The number of services is limited, one layer per time seems to cost too much.